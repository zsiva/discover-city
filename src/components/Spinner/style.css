.spinner {
  position: absolute;
  top: 20%;
  left: 50%;
  transform: translate(-50%, 0);
}

.loaderBar,
.loaderBar:before,
.loaderBar:after {
  background: #4caf50;
  -webkit-animation: load1 1s infinite ease-in-out;
  animation: load1 1s infinite ease-in-out;
  width: 1em;
  height: 4em;
}
.loaderBar {
  color: #4caf50;
  text-indent: -9999em;
  margin: 60px auto;
  position: relative;
  font-size: 11px;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation-delay: -0.16s;
  animation-delay: -0.16s;
}
.loaderBar:before,
.loaderBar:after {
  position: absolute;
  top: 0;
  content: '';
}
.loaderBar:before {
  left: -1.5em;
  -webkit-animation-delay: -0.32s;
  animation-delay: -0.32s;
}
.loaderBar:after {
  left: 1.5em;
}
@-webkit-keyframes load1 {
  0%,
  80%,
  100% {
    box-shadow: 0 0;
    height: 4em;
  }
  40% {
    box-shadow: 0 -2em;
    height: 5em;
  }
}
@keyframes load1 {
  0%,
  80%,
  100% {
    box-shadow: 0 0;
    height: 4em;
  }
  40% {
    box-shadow: 0 -2em;
    height: 5em;
  }
}
