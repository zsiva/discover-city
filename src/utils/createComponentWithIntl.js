import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import messages from '../data/messages';
import flattenMessages from '../utils/flattenMessages';

const createComponentWithIntl = (children, props = { locale: 'en-US' }) => {
  return renderer.create(
    <IntlProvider {...props} messages={flattenMessages(messages[props.locale])}>
      {children}
    </IntlProvider>,
  );
};

export default createComponentWithIntl;
