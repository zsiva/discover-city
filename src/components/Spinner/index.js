// @flow
import React from 'react';
import { type IntlShape } from 'react-intl';
import './style.css';

const Spinner = ({ text }: { text: IntlShape }) => {
  return (
    <div className="text-center spinner">
      <p>{text}</p>
      <div className="loaderBar" />
    </div>
  );
};

export default Spinner;
