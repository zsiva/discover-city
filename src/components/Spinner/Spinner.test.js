import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import createComponentWithIntl from '../../utils/createComponentWithIntl';
import Spinner from './';

describe('Spinner', () => {
  it('should be defined', () => {
    expect(Spinner).toBeDefined();
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<Spinner name="Loading..." />);
    expect(tree).toMatchSnapshot();
  });
});
