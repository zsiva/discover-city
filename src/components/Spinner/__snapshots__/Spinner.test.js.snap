// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should match snapshot 1`] = `
<div
  className="text-center spinner"
>
  <p />
  <div
    className="loaderBar"
  />
</div>
`;
